[CI] StatelessReshardIT testSourceRelocationDuringReshard failing

**Build Scans:**
- [elasticsearch-intake #42639 / part2](https://gradle-enterprise.elastic.co/s/5oyhfvjllafei)
- [elasticsearch-intake #42638 / part2](https://gradle-enterprise.elastic.co/s/qm5nosvgyljxy)
- [elasticsearch-pull-request #159427 / part-2](https://gradle-enterprise.elastic.co/s/2nxq5dvguskji)
- [elasticsearch-pull-request #159328 / part-2](https://gradle-enterprise.elastic.co/s/qm3fcn46imznw)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:stateless:internalClusterTest" --tests "org.elasticsearch.xpack.stateless.reshard.StatelessReshardIT.testSourceRelocationDuringReshard" -Dtests.seed=B17748C470D1364B -Dtests.locale=pt-MO -Dtests.timezone=America/Godthab -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.stateless.reshard.StatelessReshardIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testSourceRelocationDuringReshard'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: safeGet: listener was not completed within the timeout
```

**Issue Reasons:**
- [main] 4 failures in test testSourceRelocationDuringReshard (0.7% fail rate in 590 executions)
- [main] 2 failures in step part2 (2.4% fail rate in 83 executions)
- [main] 2 failures in step part-2 (0.5% fail rate in 439 executions)
- [main] 2 failures in pipeline elasticsearch-intake (2.4% fail rate in 83 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (0.5% fail rate in 442 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -778,18 +778,12 @@ tests:
 - class: org.elasticsearch.xpack.esql.expression.predicate.operator.arithmetic.DivTests
   method: testCrankyEvaluateBlockWithoutNulls {TestCase=<long>, <dense_vector>}
   issue: https://github.com/elastic/elasticsearch/issues/152795
-- class: org.elasticsearch.xpack.stateless.reshard.StatelessReshardIT
-  method: testSourceRelocationDuringReshard
-  issue: https://github.com/elastic/elasticsearch/issues/152800
 - class: org.elasticsearch.xpack.esql.CsvIT
   method: test {csv-spec:inlinestats.groupByMultipleRenamedColumns_AndTwoExpressions}
   issue: https://github.com/elastic/elasticsearch/issues/152801
 - class: org.elasticsearch.packaging.test.PackageUpgradeTests
   method: test12SetupBwcVersion
   issue: https://github.com/elastic/elasticsearch/issues/152818
-- class: org.elasticsearch.xpack.stateless.reshard.StatelessReshardMixedOperationsIT
-  method: testMixedOperationsDuringSplitWithDisruption
-  issue: https://github.com/elastic/elasticsearch/issues/151090
 - class: org.elasticsearch.test.rest.yaml.RcsCcsCommonYamlTestSuiteIT
   method: initializationError
   issue: https://github.com/elastic/elasticsearch/issues/152849

--- a/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/reshard/StatelessReshardIT.java
+++ b/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/reshard/StatelessReshardIT.java
@@ -83,6 +83,7 @@ import org.elasticsearch.common.Priority;
 import org.elasticsearch.common.blobstore.BlobContainer;
 import org.elasticsearch.common.blobstore.OperationPurpose;
 import org.elasticsearch.common.blobstore.support.BlobMetadata;
+import org.elasticsearch.common.settings.Setting;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.common.util.concurrent.ThreadContext;
 import org.elasticsearch.core.CheckedRunnable;
@@ -4600,7 +4601,15 @@ public class StatelessReshardIT extends AbstractStatelessPluginIntegTestCase {
         plugins.add(EncryptionPlugin.class);
         plugins.add(EsqlPlugin.class);
         plugins.add(TestTelemetryPlugin.class);
+        plugins.add(AddSettingPlugin.class);
         return plugins;
+    }
+
+    public static class AddSettingPlugin extends Plugin {
+        @Override
+        public List<Setting<?>> getSettings() {
+            return List.of(SplitTargetService.START_SPLIT_RETRY_TIMEOUT);
+        }
     }
 
     @Override
@@ -4610,7 +4619,8 @@ public class StatelessReshardIT extends AbstractStatelessPluginIntegTestCase {
             // when we start re-splitting bulk requests.
             .put(TransportReplicationAction.REPLICATION_RETRY_TIMEOUT.getKey(), "60s")
             // These tests are carefully set up and do not hit the situations that the delete unowned grace period prevents.
-            .put(RESHARD_SPLIT_DELETE_UNOWNED_GRACE_PERIOD.getKey(), TimeValue.ZERO);
+            .put(RESHARD_SPLIT_DELETE_UNOWNED_GRACE_PERIOD.getKey(), TimeValue.ZERO)
+            .put(SplitTargetService.START_SPLIT_RETRY_TIMEOUT.getKey(), TimeValue.timeValueSeconds(5));
     }
 
     @Override

--- a/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/reshard/StatelessReshardMixedOperationsIT.java
+++ b/x-pack/plugin/stateless/src/internalClusterTest/java/org/elasticsearch/xpack/stateless/reshard/StatelessReshardMixedOperationsIT.java
@@ -46,7 +46,6 @@ import static org.elasticsearch.cluster.routing.allocation.decider.MaxRetryAlloc
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertResponse;
 import static org.elasticsearch.xpack.stateless.reshard.SplitSourceService.RESHARD_SPLIT_DELETE_UNOWNED_GRACE_PERIOD;
-import static org.elasticsearch.xpack.stateless.reshard.SplitSourceService.STATE_MACHINE_RETRY_DELAY;
 import static org.hamcrest.Matchers.instanceOf;
 
 public class StatelessReshardMixedOperationsIT extends StatelessReshardDisruptionBaseIT {
@@ -97,7 +96,8 @@ public class StatelessReshardMixedOperationsIT extends StatelessReshardDisruptio
             // We should not see requests that were queued for a long time in a local cluster setup anyway.
             .put(RESHARD_SPLIT_DELETE_UNOWNED_GRACE_PERIOD.getKey(), TimeValue.timeValueMillis(100))
             // Reduce the delay between retries to speed up the test.
-            .put(STATE_MACHINE_RETRY_DELAY.getKey(), TimeValue.timeValueMillis(10));
+            .put(SplitSourceService.STATE_MACHINE_RETRY_DELAY.getKey(), TimeValue.timeValueMillis(10))
+            .put(SplitTargetService.START_SPLIT_RETRY_TIMEOUT.getKey(), TimeValue.timeValueSeconds(5));
     }
 
     @Override
@@ -108,7 +108,7 @@ public class StatelessReshardMixedOperationsIT extends StatelessReshardDisruptio
     public static class AddSettingPlugin extends Plugin {
         @Override
         public List<Setting<?>> getSettings() {
-            return List.of(STATE_MACHINE_RETRY_DELAY);
+            return List.of(SplitSourceService.STATE_MACHINE_RETRY_DELAY, SplitTargetService.START_SPLIT_RETRY_TIMEOUT);
         }
     }
 

--- a/x-pack/plugin/stateless/src/main/java/org/elasticsearch/xpack/stateless/reshard/SplitTargetService.java
+++ b/x-pack/plugin/stateless/src/main/java/org/elasticsearch/xpack/stateless/reshard/SplitTargetService.java
@@ -79,11 +79,20 @@ public class SplitTargetService {
         Setting.Property.NodeScope
     );
 
+    /// Timeout for internal retries when sending start split request to the source shard.
+    /// This setting is only configured in tests and is not registered.
+    public static final Setting<TimeValue> START_SPLIT_RETRY_TIMEOUT = Setting.positiveTimeSetting(
+        "reshard.split.target_shard_start_split_retry_timeout",
+        TimeValue.timeValueSeconds(60),
+        Setting.Property.NodeScope
+    );
+
     private static final Logger logger = LogManager.getLogger(SplitTargetService.class);
 
     private final Client client;
     private final ClusterService clusterService;
     private final ReshardIndexService reshardIndexService;
+    private final TimeValue startSplitRetryTimeout;
     private final TimeValue searchShardsOnlineTimeout;
     private final TimeValue splitStateAppliedTimeout;
 
@@ -93,6 +102,7 @@ public class SplitTargetService {
         this.client = client;
         this.clusterService = clusterService;
         this.reshardIndexService = reshardIndexService;
+        this.startSplitRetryTimeout = START_SPLIT_RETRY_TIMEOUT.get(settings);
         this.searchShardsOnlineTimeout = RESHARD_SPLIT_SEARCH_SHARDS_ONLINE_TIMEOUT.get(settings);
         this.splitStateAppliedTimeout = RESHARD_SPLIT_SPLIT_STATE_APPLIED_TIMEOUT.get(settings);
     }
@@ -481,17 +491,24 @@ public class SplitTargetService {
         }
 
         private void initiateSplitWithSourceShard(State.Clone state) {
-            var action = new InitiateSplitWithSourceShardAction(clusterService, client, split, cancelled, new ActionListener<>() {
-                @Override
-                public void onResponse(Void ignored) {
-                    advance(new State.StartSplitRpcComplete(state.recoveryListener));
-                }
+            var action = new InitiateSplitWithSourceShardAction(
+                clusterService,
+                client,
+                split,
+                cancelled,
+                startSplitRetryTimeout,
+                new ActionListener<>() {
+                    @Override
+                    public void onResponse(Void ignored) {
+                        advance(new State.StartSplitRpcComplete(state.recoveryListener));
+                    }
 
-                @Override
-                public void onFailure(Exception e) {
-                    advance(new State.FailedInRecovery(e, state.recoveryListener));
+                    @Override
+                    public void onFailure(Exception e) {
+                        advance(new State.FailedInRecovery(e, state.recoveryListener));
+                    }
                 }
-            });
+            );
             action.run();
         }
 
@@ -764,8 +781,13 @@ public class SplitTargetService {
     }
 
     // visible for tests
-    RetryableAction<Void> createInitiateSplitWithSourceShardAction(Split split, AtomicBoolean cancelled, ActionListener<Void> listener) {
-        return new InitiateSplitWithSourceShardAction(clusterService, client, split, cancelled, listener);
+    RetryableAction<Void> createInitiateSplitWithSourceShardAction(
+        Split split,
+        AtomicBoolean cancelled,
+        TimeValue retryTimeout,
+        ActionListener<Void> listener
+    ) {
+        return new InitiateSplitWithSourceShardAction(clusterService, client, split, cancelled, retryTimeout, listener);
     }
 
     private static class InitiateSplitWithSourceShardAction extends RetryableAction<Void> {
@@ -779,6 +801,7 @@ public class SplitTargetService {
             Client client,
             Split split,
             AtomicBoolean cancelled,
+            TimeValue retryTimeout,
             ActionListener<Void> listener
         ) {
             super(
@@ -788,7 +811,7 @@ public class SplitTargetService {
                 TimeValue.timeValueSeconds(5),
                 // If we haven't made progress in this much time, fail back to allocator.
                 // Maybe the source shard is assigned to a different node now.
-                TimeValue.timeValueSeconds(60),
+                retryTimeout,
                 listener,
                 clusterService.threadPool().generic()
             );

--- a/x-pack/plugin/stateless/src/test/java/org/elasticsearch/xpack/stateless/reshard/SplitTargetServiceTests.java
+++ b/x-pack/plugin/stateless/src/test/java/org/elasticsearch/xpack/stateless/reshard/SplitTargetServiceTests.java
@@ -171,7 +171,7 @@ public class SplitTargetServiceTests extends ESTestCase {
         var threadPool = new ThreadPool() {
             @Override
             public long relativeTimeInMillis() {
-                return time.getAndAdd(10 * 1000);
+                return time.getAndAdd(1000);
             }
 
             @Override
@@ -214,18 +214,23 @@ public class SplitTargetServiceTests extends ESTestCase {
         };
 
         var sts = new SplitTargetService(Settings.EMPTY, failingClient, clusterService, reshardIndexService);
-        var action = sts.createInitiateSplitWithSourceShardAction(dummySplit(), new AtomicBoolean(false), new ActionListener<>() {
-            @Override
-            public void onResponse(Void unused) {
-                fail("Should never happen");
-            }
+        var action = sts.createInitiateSplitWithSourceShardAction(
+            dummySplit(),
+            new AtomicBoolean(false),
+            TimeValue.timeValueMillis(5000),
+            new ActionListener<>() {
+                @Override
+                public void onResponse(Void unused) {
+                    fail("Should never happen");
+                }
 
-            @Override
-            public void onFailure(Exception e) {
-                // Should be 6 given the custom time implementation (6 times by 10 seconds exceeds 60 seconds timeout).
-                assertEquals(6, requests.get());
+                @Override
+                public void onFailure(Exception e) {
+                    // Should be 5 given the custom time implementation (5 times by 1000 millis exceeds 5000 millis timeout).
+                    assertEquals(5, requests.get());
+                }
             }
-        });
+        );
 
         action.run();
     }

